sklearn/metrics/pairwise.py

```diff
@@ -2285,26 +2285,31 @@ def pairwise_distances(
 ):
     """Compute the distance matrix from a vector array X and optional Y.
 
-    This method takes either a vector array or a distance matrix, and returns
+    This method takes one or two vector arrays or a distance matrix, and returns
     a distance matrix.
-    If the input is a vector array, the distances are computed.
-    If the input is a distances matrix, it is returned instead.
+
+    - If `X` is a vector array, of shape (n_samples_X, n_features), and:
+
+      - `Y` is `None` and `metric` is not 'precomputed', the pairwise distances
+        between `X` and itself are computed.
+      - `Y` is a vector array of shape (n_samples_Y, n_features), the pairwise
+        distances between `X` and `Y` is returned.
```

Comment:
It's not consistent that we say `computed` for one and `returned` for the other. Let's use `returned` for both. We should also update that in pairwise_kernels